app/controllers/autograders_controller.rb

```diff
@@ -25,15 +25,32 @@ def edit
   action_auth_level :update, :instructor
   def update
     flash[:info] = "Saved!" if @autograder.update(autograder_params)
+		upload
     redirect_to([:edit, @course, @assessment, :autograder]) && return
   end
-
+	
   action_auth_level :destroy, :instructor
   def destroy
     flash[:info] = "Destroyed!" if @autograder.destroy
     redirect_to([:edit, @course, @assessment]) && return
   end
 
+  action_auth_level :upload, :instructor
+  def upload
+    uploaded_makefile = params[:autograder][:makefile]
+	  uploaded_tar = params[:autograder][:tar]
+	  if not uploaded_makefile.nil?
```

Comment:
don't use not. Instead use `!`.

and even better, use `unless` instead (`unless condition` is the same as `if !condition`)